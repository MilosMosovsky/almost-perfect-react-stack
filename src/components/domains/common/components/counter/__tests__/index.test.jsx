import * as React from "react";
import { cleanup, render, fireEvent } from "react-testing-library";
import Icon from "../";

afterEach(cleanup);

test("init", () => {
  const { container } = render(<Icon />);

  expect(container).toMatchSnapshot();
});

test("change counter on click", () => {
  const { container, getByText } = render(<Icon />);

  const Button = getByText("Increment");
  fireEvent.click(Button);

  expect(container).toMatchSnapshot();
});
