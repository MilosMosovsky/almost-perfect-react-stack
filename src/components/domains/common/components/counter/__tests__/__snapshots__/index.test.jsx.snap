// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change counter on click 1`] = `
<div>
  <button>
    Increment
  </button>
  Counter is: 
  1
</div>
`;

exports[`init 1`] = `
<div>
  <button>
    Increment
  </button>
  Counter is: 
  0
</div>
`;
